sunpy/map/mapsequence.py

```diff
@@ -81,6 +81,16 @@ def __len__(self):
         """Return the number of maps in a mapsequence."""
         return len(self.maps)
 
+    def __repr__(self):
+        names = []
```

Comment:
This is cleaner to implement as a list comprehension instead of a loop.  You'll get a `list` with duplicate entries, of course, but you can drop the duplicates easily by converting to a `set`.